src/transformStateWithClones.js

```diff
@@ -7,7 +7,47 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateHistory = [];
+
+  let curentState = { ...state };
+
+  for (const action of actions) {
+    const { type = null, extraData = {}, keysToRemove = [] } = action;
+
+    switch (type) {
+      case 'addProperties':
+        addProperties(curentState, extraData);
+        break;
+      case 'removeProperties':
+        removeProperties(curentState, keysToRemove);
+        break;
+      case 'clear':
+        clear(curentState);
+        break;
+    }
+
+    stateHistory.push(curentState);
```

Comment:
You are directly pushing 'curentState' to 'stateHistory' which will result in all entries pointing to the same object reference. Instead, you should push a copy of 'curentState' to avoid mutation issues.